### PR TITLE
Support Curl options

### PR DESCRIPTION
**Pull request by sy-records**
close swoole/swoole-src#3828

- [x] CURLOPT_COOKIEJAR
- [x] CURLOPT_COOKIEFILE
- [x] CURLOPT_NOPROXY
- [x] CURLOPT_CERTINFO
- [x] CURLOPT_HEADEROPT
- [x] CURLOPT_PROXYHEADER
- [x] CURLOPT_KEYPASSWD
- [x] CURLOPT_SSLCERTPASSWD
- [x] CURLOPT_SSLKEYPASSWD

### CODE REVIEW

File: src/core/Curl/Handler.php [hunk 1]
```diff
@@ -125,6 +125,10 @@ final class Handler
 
     private $closed = false;
 
+    private $cookieJar = '';
+
+    private $cookieFile = '';
```
Comment (deminy) [inline src/core/Curl/Handler.php:130]: The new property `$cookieFile` is not used anywhere.

File: src/core/Curl/Handler.php [hunk 2]
```diff
@@ -811,6 +828,14 @@ private function execute()
             }
         }
 
+        if ($this->cookieJar && $this->cookieJar !== '') {
```
Comment (deminy) [inline src/core/Curl/Handler.php:831]: According to [the doc](https://curl.se/libcurl/c/CURLOPT_COOKIEJAR.html), `Specify "-" as filename to instead have the cookies written to stdout.`.

Maybe we can add a TODO and temporarily fail it directly.

**Review by deminy — CHANGES_REQUESTED** [review]

File: src/core/Curl/Handler.php (hunk 2 above)
Comment (sy-records) [reply src/core/Curl/Handler.php]: done,

File: src/core/Curl/Handler.php [hunk 3]
```diff
@@ -523,6 +535,18 @@ private function setOption(int $opt, $value): bool
             case CURLOPT_COOKIE:
                 $this->headers['Cookie'] = $value;
                 break;
+            case CURLOPT_COOKIEJAR:
+                if ($value === '-') {
+                    trigger_error('swoole_curl_setopt(): CURLOPT_COOKIEJAR is not supported written cookies to stdout', E_USER_WARNING);
+                    return false;
```
Comment (twose) [inline src/core/Curl/Handler.php:541]: 这里用STDOUT会有啥问题吗

Comment (sy-records) [reply src/core/Curl/Handler.php]: 没，改了

File: src/core/Curl/Handler.php [hunk 4]
```diff
@@ -483,6 +492,9 @@ private function setOption(int $opt, $value): bool
                     $this->headers[$headerName] = $headerValue;
                 }
                 break;
+            case CURLOPT_HEADEROPT:
```
Comment (twose) [inline src/core/Curl/Handler.php:495]: 啥也没有, 是否应该移到上面的Ignore options

**Review by twose — CHANGES_REQUESTED** [review]

File: src/core/Curl/Handler.php (hunk 4 above)
Comment (sy-records) [reply src/core/Curl/Handler.php]: 改了，写的时候看都是header的就放一起了 🤣

**Pull request merged**